functions.cpp

```diff
@@ -408,15 +408,19 @@ namespace Sass {
       HSL hsl_color = rgb_to_hsl(rgb_color->r(),
                                  rgb_color->g(),
                                  rgb_color->b());
-      //Check saturation is not negative before saturate it
-      double hslcolorS = hsl_color.s;
+
+      double hslcolorS = hsl_color.s + amount->value();
```

Comment:
@QuLogic you were right, though the comment and the way it was done before was a bit misleading. Basically if you desaturate a colour with 1% saturation by 10% it would wrap around and the bug would appear again.